Client/Application/AppDelegate.swift

```diff
@@ -137,7 +137,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     func applicationDidEnterBackground(application: UIApplication) {
         self.profile?.syncManager.endTimedSyncs()
 
-        let taskId = application.beginBackgroundTaskWithExpirationHandler { _ in }
+        var taskId: UIBackgroundTaskIdentifier = 0
+        taskId = application.beginBackgroundTaskWithExpirationHandler { _ in
+            log.warning("Running out of background time, but we have a profile shutdown pending.")
+            application.endBackgroundTask(taskId)
+        }
         self.profile?.shutdown()
```

Comment:
Really, I think that this should be in a background thread rather than running on the main thread:
`let priority = DISPATCH_QUEUE_PRIORITY_BACKGROUND
dispatch_async(dispatch_get_global_queue(priority, 0)) {
	 self.profile?.shutdown()
         application.endBackgroundTask(taskId)
}`
otherwise the task we are running is not in the background and won't be governed by the `UIBackgroundTaskIdentifier`